glommio/rusturing.c

```diff
@@ -0,0 +1,314 @@
+#include "liburing.h"
+
+extern inline int rust_io_uring_opcode_supported(struct io_uring_probe *p, int op)
+{
+    return io_uring_opcode_supported(p, op);
+}
+
+extern inline void rust_io_uring_cq_advance(struct io_uring *ring, unsigned nr)
+{
+    io_uring_cq_advance(ring, nr);
+}
+
+extern inline void rust_io_uring_cqe_seen(struct io_uring *ring, struct io_uring_cqe *cqe)
+{
+    io_uring_cqe_seen(ring, cqe);
+}
+
+extern inline void rust_io_uring_sqe_set_data(struct io_uring_sqe *sqe, void *data)
+{
+    io_uring_sqe_set_data(sqe, data);
+}
+
+extern inline void *rust_io_uring_cqe_get_data(struct io_uring_cqe *cqe)
+{
+    return io_uring_cqe_get_data(cqe);
+}
+
+extern inline void rust_io_uring_sqe_set_flags(struct io_uring_sqe *sqe, unsigned flags)
+{
+    io_uring_sqe_set_flags(sqe, flags);
+}
+
+extern inline void rust_io_uring_prep_rw(int op,
+                                         struct io_uring_sqe *sqe,
+                                         int fd,
+                                         const void *addr,
+                                         unsigned len,
+                                         __u64 offset)
+{
+    io_uring_prep_rw(op, sqe, fd, addr, len, offset);
+}
+
+extern inline void rust_io_uring_prep_splice(struct io_uring_sqe *sqe,
+					int fd_in, int64_t off_in,
+					int fd_out, int64_t off_out,
+					unsigned int nbytes,
+					unsigned int splice_flags)
+{
+    io_uring_prep_splice(sqe, fd_in, fd_out, off_in, off_out, nbytes, splice_flags);
```

Comment:
Thanks for the review! 

I would prefer not to touch this for now because we don't use `splice`. That said, if uring-sys keeps being unmaintained then maybe we will. Let's leave that on hold for a while.